src/page-templates/content-type/utils.ts

```diff
@@ -156,10 +158,49 @@ export const getFilters = async (contentType: PillCategory) => {
                         type: l1.type,
                         name: l1.name,
                         subItems: [l2FilterItem],
-                        count: category === contentType ? 1 : 0,
+                        count: allFilters || type === contentType ? 1 : 0,
                     };
                     return filterItems.push(l1FilterItem);
                 }
+            } else if (tag.type === 'CodeLevel') {
+                // Basically repeating the logic for L2s, can probably be separated and reused.
```

Comment:
Similar to the comment above regarding optimization, I would capture this work in a ticket to revisit.